src/durable-objects/SessionFileTracker.ts

```diff
@@ -0,0 +1,284 @@
+import { DurableObject } from "cloudflare:workers";
+
+export interface FileMetadata {
+  fileKey: string;
+  fileName: string;
+  fileSize: number;
+  status: "uploading" | "uploaded" | "parsing" | "parsed" | "error";
+  uploadedAt: string;
```

Comment:
question (not for this PR): do you ever see us updating metadata for a file without changing the file? have you thought about that flow? (i really dk if necessary, def not v1, but interested in your thoughts on options/possibilities)

Reply:
in which case i feel like we should have `createdAt` and `updatedAt` (rather than the current `uploadedAt`)

Reply:
do you know how we "migrate" data within a durable object? if we add new metadata fields, for example, do we just start writing them and ignore reads from the old fields? do we ever need to clean up? how? (i'll ask chat too)

Reply:
> do you ever see us updating metadata for a file without changing the file?

yeah, I think so. I envisioned the tags as eventually being searchable so you can start a campaign off of a group of PDFs all at once if you have a tag across all of them. So updating the tags specifically seems likely.

Reply:
> do you know how we "migrate" data within a durable object? if we add new metadata fields, for example, do we just start writing them and ignore reads from the old fields? do we ever need to clean up? how? (i'll ask chat too)

not sure on the answer to this rn but I can research
@deloreyj might know